cmd/server/main.go

```diff
@@ -117,7 +119,9 @@ func main() {
 	pb.RegisterKeyManagementServiceServer(s, kmsServer)
 
 	klog.InfoS("Listening for connections", "addr", listener.Addr().String())
-	go s.Serve(listener)
+	go func() {
+		_ = s.Serve(listener)
```

Comment:
Given this is critical for the plugin, in case of error we should log the error and exit here!

```suggestion
		if err := s.Serve(listener); err != nil {
		        klog.ErrorS(err, "failed to serve")
		        os.Exit(1)
		 }
```